tests/unit/minio_test.py

```diff
@@ -28,23 +28,23 @@
 class ValidBucketName(TestCase):
     @raises(InvalidBucketError)
     def test_bucket_name(self):
-        is_valid_bucket_name('bucketName')
+        is_valid_bucket_name('bucketName', False)
```

Comment:
@ebozduman, unit tests won't pass here, you need to pass True instead of False in your change in this file

Reply:
(Run this `bash tests/unit_test.sh`)

Reply:
The fix is to introduce a relaxed validation capability for all apis, except `make_bucket` api. So, instead of changing to strict validation, I changed the tests.